Lib/Src/error_log.cpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file error_log.cpp
+ */
+
+/*
+ *  Created on: Feb 12, 2022
+ *      Author: niuslar
+ */
+
+#include "error_log.h"
+
+/**
+ * @brief Constructor
+ * @param Pointer to CUartCom object
+ * @param module_name Identifier that will be added to the message
+ */
+CLog::CLog(CUartCom *error_uart, const std::string module_name)
+    : mp_error_uart(error_uart),
+      m_module_name(module_name)
+{
+}
+
+/**
+ * @brief change log level to change what is sent
+ * @param log_level:
+ * 			LOG_ERROR = will show only errors
+ * 			LOG_WARNING = will show error and warnings
+ * 			LOG_INFO = will show everything
+ */
+void CLog::setLogLevel(log_level_t log_level)
+{
+    m_log_level = log_level;
```

Comment:
Can someone force log_level to be a value outside enum type?

Reply:
Added a case to the log() method in case someone does that.